app/controllers/users_controller.rb

```diff
@@ -0,0 +1,31 @@
+class UsersController < ApplicationController
+  load_and_authorize_resource
+
+  skip_before_action :authenticate_user!, only: %i[index show]
+
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  # Create an index action
+  def index
+    @users = User.all
+  end
+
+  # Create a show action
+  def show
+    @user = User.find(params[:id])
+    @categories = @user.categories
+    # Other logic as needed
+    p @user
+    p @categories
+  end
+
+  private
+
+  def not_found
+    render file: "#{Rails.root}/public/404.html", layout: false, status: 404
+  end
+
+  def user_params
```

Comment:
![image](https://github.com/diarisdiakite/my_account_app/assets/78251927/dc7c9361-075a-411a-96fb-6b810a3062d2)

Please note that every user's details are private and only accessible when that particular user login.

I see that in your app, a logged-in user is able to access the transactions of all other users, which is wrong, please correct that 🛠